nuitka/optimizations/OptimizeBuiltinCalls.py

```diff
@@ -1000,6 +1015,188 @@ def wrapExpressionBuiltinExecCreation(
+            name="zip_call",
+            source_ref=source_ref,
+        )
+
+        zip_arg_variables = [
+            outline_body.allocateTempVariable(
+                temp_scope=None, name="zip_arg_%d" % (i + 1)
+            )
+            for i in range(len(call_args))
+        ]
+
+        statements = [
+            StatementAssignmentVariable(
+                variable=zip_arg_variable, source=call_arg, source_ref=source_ref
+            )
+            for call_arg, zip_arg_variable in zip(call_args, zip_arg_variables)
+        ]
+
+        zip_iter_variables = [
+            outline_body.allocateTempVariable(
+                temp_scope=None, name="zip_iter_%d" % (i + 1)
+            )
+            for i in range(len(call_args))
+        ]
+
+        # Add check for all argument if they are iterable
+        # statements += [
+        #     makeStatementConditional(
+        #         condition=ExpressionAttributeCheck(
+        #             object_arg=ExpressionTempVariableRef(
+        #                 variable=zip_arg_variable, source_ref=source_ref
+        #             ),
+        #             attribute_name="__iter__",
+        #             source_ref=source_ref,
+        #         ),
+        #         yes_branch=None,
+        #         # This should become the exception
+        #         # handle for TypeError
+        #         no_branch=makeRaiseExceptionExpressionFromTemplate(
+        #             exception_type="TypeError",
+        #             template="zip argument #%s must support iteration",
+        #             template_args=makeConstantRefNode(
+        #                 constant=str(i+1), source_ref=source_ref
+        #             ),
+        #             source_ref=source_ref,
+        #         ).asStatement(),
+        #         source_ref=source_ref,
+        #     )
+        #     for i, zip_arg_variable in enumerate(zip_arg_variables)
+        # ]
+
+        # Create iterators of all zip arguments.
+        assign_iter_statements = [
+            StatementAssignmentVariable(
+                variable=zip_iter_variable,
+                source=ExpressionBuiltinIter1(
+                    value=ExpressionTempVariableRef(
+                        variable=zip_arg_variable, source_ref=source_ref
+                    ),
+                    source_ref=source_ref,
+                ),
+                source_ref=source_ref,
+            )
+            for zip_iter_variable, zip_arg_variable in zip(
+                zip_iter_variables, zip_arg_variables
+            )
+        ]
+
+        # makeTryExceptSingleHandlerNode
+        statements += [
+            makeTryExceptSingleHandlerNode(
+                tried=assign_iter_statements,
+                exception_name="TypeError",
+                handler_body=makeRaiseExceptionExpressionFromTemplate(
+                    exception_type="TypeError",
+                    template="zip argument #%s must support iteration",
```

Comment:
Why use a template with constant arguments, seems the value is known and `%d` would be better suited too, either way.

Reply:
There may be a function that uses no template, but similarily does it.